### PR TITLE
BRICK-989: Mask locksmithd

### PR DESCRIPTION
**Pull request by jaime-murillo**

### CODE REVIEW

**Review by mordr — APPROVED** [review]

**Comment by amattyang** [comment]
I thought our convention for this repo is to modify cl.config file then generate ignition file(*.ign) from it? 

cc: @ahawtho

**Comment by jaime-murillo** [comment]
Ok, I'll update. is there a link for other conventions?

**Comment by amattyang** [comment]
@jaimemurillo16 : let's hold off on the change until if @ahawtho has any input on this. Also, I don't think we have a document now. It's something I remember but I might be wrong.

**Comment by jaime-murillo** [comment]
@amattyang , I think you are right as the ignition config is **not** meant to be [edited](https://coreos.com/os/docs/latest/clc-examples.html).

**Review by mordr — APPROVED** [review]
missed the cl.conf, I think this should be ok

**Comment by mordr** [comment]
@amattyang is correct, we modify the `cl.conf` file, and run the `build.sh` to generate the updated `config.ign` file in `bricks/config.ign`.

**Review by mordr — DISMISSED** [review]

File: cl.conf
Comment (amattyang) [inline cl.conf:53]: It seems we can use cl syntax to disable locksmithd.
https://coreos.com/os/docs/latest/update-strategies.html
```
systemd:
  units:
    - name: locksmithd.service
      mask: true
```

Also, if we do 'vagrant halt' then 'vagrant up', does it enable a disabled locksmithd service?

Comment (mordr) [reply cl.conf:53]: I think the `RemainAfterExit=true` is supposed to keep the state throughout the vm lifetime ?

Comment (jaime-murillo) [reply cl.conf:53]: @amattyang, kops doesnt allow masking a service in this method at the moment.  You can disable, but not mask.  I was keeping it the same as the kops method. Could switch it up here.

No, either way on 'vagrant halt' then 'vagrant up' locksmithd will not be started, as this masks the service and doesnt just disable it.  Recall disable only stops the service, until the next reboot.  

@mordr, I dont think it matters in this case. RemainAfterExit is how systemd treats the status of the service after ExecStart exits. If true, then after the ExecStart, the status of the service is kept as active. If left at the default (false) then after ExecStart exits, on a oneshot, the status is set to inactive.

RemainAfterExit is useful if you want to keep the oneshot service active, so you can **stop** the service at a later date, which could run an **ExecStop**. i.e in case you wanted to revert the masking of locksmithd.

Something like
`ExecStop=/usr/bin/systemctl unmask --now locksmithd.service`
would unmask locksmithd when __systemctl stop mask-locksmithd__ is executed

Comment (mordr) [reply cl.conf:53]: ok, thanks.

**Comment by mordr** [comment]
I think the cl.conf changes for masking locksmithd lgtm. I am going to merge this one because I saw data team run into the update problem yesterday as well. This should unblock anyone suffering from the coreos-vagrant auto update.

**Review by mordr — APPROVED** [review]

File: cl.conf
Comment (amattyang) [reply cl.conf:53]: @jaimemurillo16 : let's switch it up to use cl syntax if you don't mind.

nvm, it's been merged by @mordr

**Pull request merged**
